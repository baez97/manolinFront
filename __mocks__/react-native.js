function AsyncStorageClass() {
    this.setItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.getItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve("THIS_IS_THE_SAVED_TOKEN");
        })
    });

    this.removeItem = jest.fn((key) => {
        return  new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.removeItem.mockReturnValueOnce(new Promise((res,rej) => {
        rej({message: "THIS IS AN ERROR"});
    }))
}

module.exports.AsyncStorage = new AsyncStorageClass();