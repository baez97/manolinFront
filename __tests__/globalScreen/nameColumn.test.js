import React from 'react';
import renderer from 'react-test-renderer';
import NameColumn from '../../components/globalComponents/nameColumn';
import mockedNurse from '../../mockedData/mockedNurse';
import {Â Text } from 'react-native';

jest.mock("../../styles/layoutStyle")
jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

describe("NameColumn tests", () => {

    // MOCKING DATA
    const nurses = [];
    for ( let i = 0; i < 11; i++ ) 
        nurses.push(mockedNurse);

    it("Matches the Snapshot", () => {
        // RENDERING
        const view         = <NameColumn nurses={nurses} />;
        const rendered     = renderer.create(view);
        const renderedJSON = rendered.toJSON();

        // ASSERTION
        expect( renderedJSON ).toMatchSnapshot();
    });

    it("Has a FlatList with the expected items", () => {
        // EXPECTED OUTPUT
        const expectedLength = nurses.length;

        // RENDERING
        const nC       = new NameColumn({ nurses });
        const rendered = nC.render();

        // OBTAINING THE ITEMS
        const list  = rendered.props.children;
        const items = list.props.data;

        // ASSERTION
        expect(items.length).toBe(expectedLength);
        items.forEach( i => {
            expect( i ).toBe('Ana');
        });
    });

    it("RenderItem method works properly in the FlatList", () => {
        // EXPECTED OUTPUT
        // const expectedView = 
        //     ( <Text style={{
        //         allowFontScaling={false},
        //         fontFamily: 'big-noodle-titling',
        //         fontSize: 30,
        //         paddingBottom: 10,}}>
        //         { mockedNurse.name }
        //     </Text> );

        // RENDERING
        const nC       = new NameColumn({ nurses });
        const rendered = nC.render();

        // TESTED METHOD EXECUTION
        const list       = rendered.props.children;
        const actualView = list.props.renderItem({item: mockedNurse.name});

        // ASSERTION
        // expect( actualView ).toEqual(expectedView);
        expect( actualView ).toMatchSnapshot();
    })
});