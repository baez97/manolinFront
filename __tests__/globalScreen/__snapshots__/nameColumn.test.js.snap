// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameColumn tests Matches the Snapshot 1`] = `
<View
  style={
    Object {
      "marginRight": 20,
      "marginTop": 45,
    }
  }
/>
`;

exports[`NameColumn tests RenderItem method works properly in the FlatList 1`] = `
<Component
  allowFontScaling={false}
  style={
    Object {
      "fontFamily": "big-noodle-titling",
      "fontSize": 30,
      "paddingBottom": 10,
    }
  }
>
  Ana
</Component>
`;
